app/services/solidus_square/create_payment_service.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module SolidusSquare
+  class CreatePaymentService < Base
+    attr_reader :source_id, :order, :payment_method_id
+
+    def initialize(source_id:, order:, payment_method_id:)
+      @source_id = source_id
+      @order = order
+      @payment_method_id = payment_method_id
+
+      super()
+    end
+
+    def call
+      order.next! if order.delivery?
+      complete_payment!
+    end
+
+    private
+
+    def complete_payment!
+      payment = create_payment!
+      payment.complete! if payment.source.captured?
```

Comment:
It won't happen because the payment will be captured only after order complete.

Reply:
https://github.com/nebulab/solidus_square/pull/49/commits/895e71c2f8fa90e07e6277827b947919170b1292